frontend/__tests__/components/AddBookModal/add-book-modal.test.js

```diff
@@ -0,0 +1,47 @@
+import React from "react";
+import { render, screen, userEvent } from "../../../test-utils";
+import AddBookModal from "../../../components/AddBookModal/AddBookModal";
+
+describe("Add Book Modal", () => {
+  const onCloseMock = jest.fn();
+  it("should render add book modal component", () => {
+    render(
+      <div id="app-root">
+        <AddBookModal />
+      </div>
+    );
+    expect(screen.getByText("Add New Book")).toBeInTheDocument();
+    expect(screen.getByText("Cancel")).toBeInTheDocument();
+    expect(screen.getByText("Add Book")).toBeInTheDocument();
+  });
+  it("should call onClose when the cancel button is clicked", async () => {
+    render(
+      <div id="app-root">
+        <AddBookModal onClose={onCloseMock} />
+      </div>
+    );
+    const cancelButton = screen.getByText("Cancel");
+    expect(cancelButton).toBeInTheDocument();
+    await userEvent.click(cancelButton);
```

Comment:
Please look into userEvent.

Reply:
I need the await currently because the click function in user-event@14 returns a promise so I need the interpreter to wait for the asynchronous code to finish.